specification/dartLangSpec.tex

```diff
@@ -15945,23 +15951,46 @@ \subsection{Lexical Lookup}
   It is a compile-time error if $\ell$ does not have access to \THIS{}
   (\ref{classes}).
 \item
-  When $\ell$ has access to \THIS{}
-  it is a compile-time error if $\ell$ occurs in
-  a class \DefineSymbol{C} and
-  the interface of $C$ does not have a member named $n$;
-  and it is a compile-time error if $\ell$ occurs in
-  an extension \DefineSymbol{E},
-  and the interface of the \ON{} type of $E$ does not have a member named $n$.
-  \commentary{%
-    So it is an error if there is no member with the right basename at all,
-    and also if there is one such member,
-    but it is a non-setter and we are looking for a setter,
-    or vice versa.%
-  }
+  Consider the case where $\ell$ has access to \THIS,
+  and $\ell$ occurs in a class \DefineSymbol{C}.
+  A compile-time error occurs in each of the following two situations:
+  \begin{itemize}
+  \item
+    The interface of $C$ does not have a member named $n$,
+    but it does have a member with basename \id.
+  \item
+    The interface of $C$ does not have a member with basename \id,
+    and there is no accessible extension $E$,
+    such that $C$ matches the \ON{} type of $E$,
+    and $E$ has an instance member named $n$.
+  \end{itemize}
+\item
+  \BlindDefineSymbol{E, T}%
+  Consider the case where $\ell$ has access to \THIS,
+  and $\ell$ occurs in an extension $E$ with \ON{} type $T$.
+  A compile-time error occurs in the following two situations:
+  \begin{itemize}
+  \item
+    The interface of $T$ does not have a member named $n$,
+    but it does have a member with basename \id.
+  \item
+    The interface of $T$ does not have a member with basename \id,
+    and there is no accessible extension $E_1$,
+    such that $T$ matches the \ON{} type of $E_1$,
+    and $E_1$ has an instance member named $n$.
+  \end{itemize}
 \end{itemize}
 \vspace{-2ex}
 \EndCase
 
+\commentary{%
+So it is an error if there is no member with the right basename,
+and there is also no applicable extension member with the right basename.
+It is also an error if there is one such member,
```

Comment:
"one or more such members"?

Reply:
I think we can't actually have more than one: We can have two members with the given basename, but then we must have both a getter and a setter. ;-)

Reply:
So there's an implicit assumption that this code is non-erroneous?  e.g. doesn't have duplicate definitions?